modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-builder/src/main/java/com/liferay/dynamic/data/mapping/form/builder/internal/servlet/DDMDataProviderInstancesServlet.java

```diff
@@ -105,8 +104,7 @@ private JSONArray _getDataProviderInstancesJSONArray(
 				scopeGroupId = scopeGroup.getLiveGroupId();
 			}
 
-			long[] groupIds = _portal.getCurrentAndAncestorSiteGroupIds(
-				scopeGroupId);
+			long[] groupIds = {scopeGroupId};
 
 			int start = ParamUtil.getInteger(
 				httpServletRequest, "start", QueryUtil.ALL_POS);
```

Comment:
@icgsbr we should keep the current behavior to avoid regression bugs.. I was thinking about something like:

```
long scopeGroupId = ParamUtil.getLong(
	httpServletRequest, "scopeGroupId",
	themeDisplay.getScopeGroupId());

Group scopeGroup = themeDisplay.getScopeGroup();

if (scopeGroup.isStagingGroup()) {
	scopeGroupId = scopeGroup.getLiveGroupId();
}

int start = ParamUtil.getInteger(
	httpServletRequest, "start", QueryUtil.ALL_POS);
int end = ParamUtil.getInteger(
	httpServletRequest, "end", QueryUtil.ALL_POS);

DataProviderInstanceNameComparator
	dataProviderInstanceNameComparator =
		new DataProviderInstanceNameComparator(true);

List<DDMDataProviderInstance> ddmDataProviderInstances =
	ListUtil.concat(
		_ddmDataProviderInstanceService.search(
			themeDisplay.getCompanyId(), new long[] {scopeGroupId},
			null, start, end, dataProviderInstanceNameComparator),
		_ddmDataProviderInstanceService.search(
			themeDisplay.getCompanyId(),
			_portal.getAncestorSiteGroupIds(scopeGroupId), null,
			start, end, dataProviderInstanceNameComparator));
```

Could you please test to see if it works?

Thanks!

Reply:
Hi @carolmariaabb I'm still trying to do this implementation you suggested, currently I'm running into some erros, I'll try to fix them in order to be able to do some proper tests. I'll update you ASAP.

Reply:
Hi @carolmariaabb! I've completed the implementation! It works perfectly!